File: algo/src/main/java/com/hartwig/actin/algo/soc/datamodel/EvaluatedTreatment.java
package com.hartwig.actin.algo.soc.datamodel;

import java.util.List;

import com.hartwig.actin.algo.datamodel.Evaluation;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class EvaluatedTreatment {

    @NotNull
    public abstract Treatment treatment();

    @NotNull
    public abstract List<Evaluation> evaluations();

    public abstract int score();
}

File: common/src/main/kotlin/com/hartwig/actin/algo/datamodel/EvaluatedTreatment.kt
package com.hartwig.actin.algo.datamodel

data class EvaluatedTreatment(
    val treatmentCandidate: TreatmentCandidate,
    val evaluations: List<Evaluation>
) {

    fun eligible() = evaluations.none { it.result == EvaluationResult.FAIL }
}